skimage/morphology/gray.py

```diff
@@ -1,33 +1,33 @@
 """
 Grayscale morphological operations
 """
-import functools
+import warnings
 
 import numpy as np
 from scipy import ndimage as ndi
 
-from ..util import crop
-from .footprints import _footprint_is_sequence, _shape_from_sequence
+from .footprints import _footprint_is_sequence, mirror_footprint, pad_footprint
 from .misc import default_footprint
+from .._shared.utils import DEPRECATED
+
 
 __all__ = ['erosion', 'dilation', 'opening', 'closing', 'white_tophat', 'black_tophat']
 
 
-def _iterate_gray_func(gray_func, image, footprints, out):
-    """Helper to call `binary_func` for each footprint in a sequence.
+def _iterate_gray_func(gray_func, image, footprints, out, mode, cval):
+    """Helper to call `gray_func` for each footprint in a sequence.
 
-    binary_func is a binary morphology function that accepts "structure",
-    "output" and "iterations" keyword arguments
-    (e.g. `scipy.ndimage.binary_erosion`).
+    `gray_func` is a morphology function that accepts `footprint`, `output`,
+    `mode` and `cval` keyword arguments (e.g. `scipy.ndimage.grey_erosion`).
     """
     fp, num_iter = footprints[0]
-    gray_func(image, footprint=fp, output=out)
+    gray_func(image, footprint=fp, output=out, mode=mode, cval=cval)
```

Comment:
See https://github.com/scikit-image/scikit-image/pull/6695/commits/4a31f39bf984952c3cfbb4ee811248bfb8777ecb and https://github.com/scikit-image/scikit-image/pull/6695/commits/eb02ed77cb4e5c2ca824b17980afc2396ddde08f.